packages/backend/src/managers/playgroundV2Manager.ts

```diff
@@ -38,26 +40,62 @@ export class PlaygroundV2Manager extends Publisher<PlaygroundV2[]> implements Di
   constructor(
     webview: Webview,
     private inferenceManager: InferenceManager,
+    private taskRegistry: TaskRegistry,
   ) {
     super(webview, Messages.MSG_PLAYGROUNDS_V2_UPDATE, () => this.getPlaygrounds());
     this.#playgrounds = new Map();
     this.#conversationRegistry = new ConversationRegistry(webview);
     this.#UIDcounter = 0;
   }
 
-  async createPlayground(name: string, model: ModelInfo): Promise<void> {
+  async requestCreatePlayground(name: string, model: ModelInfo): Promise<string> {
+    const trackingId: string = getRandomString();
+    const task = this.taskRegistry.createTask('Creating Playground environment', 'loading', {
+      trackingId: trackingId,
+    });
+
+    this.createPlayground(name, model)
+      .then((playgroundId: string) => {
+        this.taskRegistry.updateTask({
```

Comment:
You mean `await this.createPlayground(name, model)` ?

Reply:
Yes as it is the pattern we use in general

Reply:
I don't see why it has an impact

Reply:
If we use `await createPlayground` here, it will wait for this function to resolve before to return `trackingId`. But this createPlayground can be long to be resolved (the time the inference server image is downloaded, the container started, etc)